android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/Worker.kt

```diff
@@ -34,3 +44,39 @@ interface Worker {
   @JvmDefault
   fun onStop() {}
 }
+
+// ---- CoroutineWorker <-> Worker adapters ----//
+
+public fun Worker.asCoroutineWorker(): CoroutineWorker = WorkerToCoroutineWorkerAdapter(this)
```

Comment:
This seems a bit weird to me. Given that we adding a whole new "Rib" coroutine worker interface, this definition within a "rib" Worker shouldn't have to know anything about CoroutineWorker

Reply:
This can surely be in a different file. It can't be in the `rib-coroutine` module, though, as `rib-base` contains `Worker` and depends on `rib-coroutine` and not the other way around. `rib-coroutine` doesn't know about `Worker`.

Reply:
Just want to add that, even though it can surely be in a different file, not sure if it *should*. 

In fact, I don't think all of the following components should be defined in a file on their own, as they currently are: `Worker`, `WorkerBinder`, `WorkerScopeProvider`, `WorkerUnbinder`. This one-class per file (or one interface per file) is reminiscent of Java, but in Kotlin it's fine (even preferable) to put all closely related components (classes, interfaces, top-level functions, etc) on a single file.

Reply:
Moved `RibCoroutineWorker` to `ribs-base` and adapters to `RibCoroutineWorker.kt`